flink-staging/flink-streaming/flink-streaming-core/src/test/java/org/apache/flink/streaming/api/complex/ComplexIntegrationTest.java

```diff
@@ -0,0 +1,903 @@
+import org.apache.flink.streaming.api.windowing.deltafunction.DeltaFunction;
+import org.apache.flink.streaming.api.windowing.helper.Count;
+import org.apache.flink.streaming.api.windowing.helper.Delta;
+import org.apache.flink.streaming.api.windowing.helper.Time;
+import org.apache.flink.streaming.api.windowing.helper.Timestamp;
+import org.apache.flink.streaming.util.RectangleClass;
+import org.apache.flink.streaming.util.TestStreamEnvironment;
+import org.apache.flink.util.Collector;
+import org.junit.Test;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+
+public class ComplexIntegrationTest implements Serializable {
+	private static final long serialVersionUID = 1L;
+	private static final long MEMORYSIZE = 32;
+
+	private static Map<String, List<String>> results = new HashMap<String, List<String>>();
+
+	@SuppressWarnings("unchecked")
+	public static List<Tuple5<Integer, String, Character, Double, Boolean>> input = Arrays.asList(
+			new Tuple5<Integer, String, Character, Double, Boolean>(1, "apple", 'j', 0.1, false),
+			new Tuple5<Integer, String, Character, Double, Boolean>(1, "peach", 'b', 0.8, true),
+			new Tuple5<Integer, String, Character, Double, Boolean>(1, "orange", 'c', 0.7, false),
+			new Tuple5<Integer, String, Character, Double, Boolean>(2, "apple", 'd', 0.5, false),
+			new Tuple5<Integer, String, Character, Double, Boolean>(2, "apple", 'e', 0.6, false),
+			new Tuple5<Integer, String, Character, Double, Boolean>(3, "peach", 'a', 0.2, true),
+			new Tuple5<Integer, String, Character, Double, Boolean>(6, "peanut", 'b', 0.1, true),
+			new Tuple5<Integer, String, Character, Double, Boolean>(7, "banana", 'c', 0.4, false),
+			new Tuple5<Integer, String, Character, Double, Boolean>(8, "peanut", 'd', 0.2, false),
+			new Tuple5<Integer, String, Character, Double, Boolean>(10, "cherry", 'e', 0.1, false),
+			new Tuple5<Integer, String, Character, Double, Boolean>(10, "plum", 'a', 0.5, true),
+			new Tuple5<Integer, String, Character, Double, Boolean>(11, "strawberry", 'b', 0.3, false),
+			new Tuple5<Integer, String, Character, Double, Boolean>(11, "orange", 'c', 0.3, true),
+			new Tuple5<Integer, String, Character, Double, Boolean>(12, "lemon", 'd', 0.9, false),
+			new Tuple5<Integer, String, Character, Double, Boolean>(13, "apple", 'e', 0.7, false),
+			new Tuple5<Integer, String, Character, Double, Boolean>(15, "lemon", 'a', 0.2, false),
+			new Tuple5<Integer, String, Character, Double, Boolean>(16, "apple", 'b', 0.8, true),
+			new Tuple5<Integer, String, Character, Double, Boolean>(16, "banana", 'c', 0.8, false),
+			new Tuple5<Integer, String, Character, Double, Boolean>(16, "orange", 'd', 0.1, false),
+			new Tuple5<Integer, String, Character, Double, Boolean>(17, "cherry", 'e', 1.0, false));
+
+//	@Test
```

Comment:
It would be great to have a quick summary (one or two lines) what the test is doing. That helps a lot understanding the code.